### PR TITLE
Run unit tests

### PR DESCRIPTION
**Pull request by DanielMcInnes**

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -65,11 +80,6 @@ message("Building VenusOS for ${CMAKE_SYSTEM_NAME}")
 message("CMAKE_CXX_FLAGS: ${CMAKE_CXX_FLAGS}")
 message("CMAKE_C_FLAGS: ${CMAKE_C_FLAGS}")
 
-option(VENUS_DESKTOP_BUILD "enable desktop build via cmake -DVENUS_DESKTOP_BUILD=ON" OFF) # Disabled by default
```
Comment (DanielMcInnes) [inline CMakeLists.txt:68]: VENUS_DESKTOP_BUILD, VENUS_WEBASSEMBLY_BUILD, and MQTT_WEBSOCKETS_ENABLED were never used for anything

Comment (chriadam) [reply CMakeLists.txt:68]: MQTT_WEBSOCKETS_ENABLED is used in src/veutil for the mqtt stuff.

Comment (chriadam) [reply CMakeLists.txt:68]: main.cpp has /main.cpp:259:#if !defined(VENUS_WEBASSEMBLY_BUILD) && !defined(VENUS_DESKTOP_BUILD) etc

Comment (DanielMcInnes) [reply CMakeLists.txt:68]: The cmake options were never used, i.e. `cmake -DVENUS_WEBASSEMBLY_BUILD=ON` doesn't do anything. The VENUS_WEBASSEMBLY_BUILD preprocessor stuff still gets set the same way, and still works

**Comment by blammit** [comment]
I like having the tests run regularly, but on every build does seem a bit much, as it adds verbose output and a lot more noise to the logs when I rebuild and run. Could this be enabled with a specific flag instead?

It definitely makes sense to run tests whenever a PR is pushed to the repo. It would be nice to do that somehow.

**Comment by DanielMcInnes** [comment]
> I like having the tests run regularly, but on every build does seem a bit much, as it adds verbose output and a lot more noise to the logs when I rebuild and run. Could this be enabled with a specific flag instead?
> 
> It definitely makes sense to run tests whenever a PR is pushed to the repo. It would be nice to do that somehow.

Updated - the unit tests no longer run every build, unless you specify `cmake -DRUN_UNIT_TESTS=ON`
The unit tests are run on the CI machine when you push changes

File: CMakeLists.txt (hunk 1 above)
Comment (chriadam) [reply CMakeLists.txt:68]: Ah!  I see

**Comment by chriadam** [comment]
Looks great, thanks.  I haven't had a chance to test this properly yet, though.  I will by Thursday for sure.

File: CMakeLists.txt [hunk 2]
```diff
@@ -1018,6 +1025,16 @@ else()
     )
 endif()
 
+if (("${RUN_UNIT_TESTS}" STREQUAL "ON") AND ("${VENUS_DESKTOP_BUILD}" STREQUAL "ON")) # Don't run unit tests when building wasm or cerbo
```
Comment (blammit) [inline CMakeLists.txt:1028]: Should it always run unit tests if RUN_UNIT_TESTS is set, regardless of VENUS_DESKTOP_BUILD? If it's set then it seems like you know what you're doing, regardless of the platform.

**Review by blammit — APPROVED** [review]
Great! Nice work on adding the github action as well.

File: CMakeLists.txt (hunk 2 above)
Comment (DanielMcInnes) [reply CMakeLists.txt]: sure, updated as suggested

**Pull request merged**